### PR TITLE
Webpack 2 compatibility

### PR DESCRIPTION
**Pull request by papandreou**

### CODE REVIEW

**Comment by papandreou** [comment]
Something's up with the file-loader.

Repro: `mocha --no-exit -g 'bundleWebpack.*discover relations|should work when combined with the asset list plugin'`

**Comment by papandreou** [comment]
`mocha -i -g bundleSystemJs` passes. Must be some interference from systemjs-builder.

**Comment by papandreou** [comment]
@guybedford, do you think there would be a way for systemjs-builder to clean up?

File: lib/transforms/bundleSystemJs.js [hunk 1]
```diff
@@ -928,9 +939,14 @@ module.exports = function (options) {
             bundleStrategyOptions.conditions = options.conditions;
         }
 
+        var globalSnapshotBeforeBundling = Object.assign({}, global);
+        _.defaults(global, systemJsGlobals);
         return bundleStrategy(builder, entryPointNames.map(function (entryPointName) {
             return { name: entryPointName };
         }), bundleStrategyOptions).then(function (bundles) {
+            Object.keys(systemJsGlobals).forEach(function (key) {
+                delete global[systemJsGlobals];
```
Comment (gustavnikolaj) [inline lib/transforms/bundleSystemJs.js]: s/systemJsGlobals/key/

Comment (papandreou) [inline lib/transforms/bundleSystemJs.js]: Whoops, that shouldn't even be there, the cleanup is handled in the `finally` block. Fixed.

**Comment by guybedford** [comment]
This seems sensible! Will definitely aim to remove leaks in subsequent versions.

**Comment by papandreou** [comment]
Landed in 3.0.0-48 / ag-b 5.0.0-38

**Pull request merged**